File: app/src/main/java/com/sayut61/hockey/ui/utils/SvgLoader.java
package com.sayut61.hockey.ui.utils;

import android.app.Activity;
import android.net.Uri;
import android.widget.ImageView;

import com.sayut61.hockey.ui.utils.SvgParser;

public class SvgLoader {

    private static SvgLoader instances;

    private SvgParser svgParser;
    private Activity activity;

    public static SvgLoader pluck() {
        if (instances == null)
            instances = new SvgLoader();
        return instances;
    }

    public SvgLoader with(Activity activity){
        this.activity = activity;
        svgParser = new SvgParser(activity);
        return instances;
    }

    public void close(){
        svgParser.clearCache();
    }

    public SvgLoader setPlaceHolder(int placeHolderLoading, int placeHolderError){
        svgParser.setPlaceHolder(placeHolderLoading, placeHolderError);
        return instances;
    }

    public SvgLoader load(String url, ImageView imageView){
        Uri uri = Uri.parse(url);
        svgParser.loadImage(uri, imageView);
        return instances;
    }

    public SvgLoader load(Uri uri, ImageView imageView){
        svgParser.loadImage(uri, imageView);
        return instances;
    }
}

File: app/src/main/java/com/sayut61/hockey/ui/utils/SvgLoader.kt
package com.sayut61.hockey.ui.utils

import android.widget.ImageView
import androidx.appcompat.widget.AppCompatImageView
import coil.ImageLoader
import coil.decode.SvgDecoder
import coil.load
import coil.request.ImageRequest
import java.util.*

suspend fun ImageView.loadSvg(url: String) {
    if (url.lowercase(Locale.ENGLISH).endsWith("svg")) {
        val imageLoader = ImageLoader.Builder(context)
            .componentRegistry {
                add(SvgDecoder(context))
            }
            .build()
        val request = ImageRequest.Builder(context)
            .data(url)
            .target(this)
            .build()
        imageLoader.execute(request)
    } else {
        load(url)
    }
}


//suspend fun AppCompatImageView.loadSvgOrOthers(myUrl: String?) {
//    myUrl?.let {
//        if (it.lowercase(Locale.ENGLISH).endsWith("svg")) {
//            val imageLoader = ImageLoader.Builder(this.context)
//                .componentRegistry {
//                    add(SvgDecoder(this@loadSvgOrOthers.context))
//                }
//                .build()
//            val request = ImageRequest.Builder(this.context)
//                .data(it)
//                .target(this)
//                .build()
//            imageLoader.execute(request)
//        } else {
//            this.load(myUrl)
//        }
//    }
//}